lib/ex_plasma/transaction/type/payment_v1.ex

```diff
@@ -1,150 +1,251 @@
 defmodule ExPlasma.Transaction.Type.PaymentV1 do
-  @moduledoc false
+  @moduledoc """
+  Internal representation of a raw payment transaction done on Plasma chain.
 
-  @behaviour ExPlasma.Transaction
+  This module holds the representation of a "raw" transaction, i.e. without signatures nor recovered input spenders
+  """
+  alias ExPlasma.Crypto
+  alias ExPlasma.Output
+  alias ExPlasma.Transaction.TypeMapper
 
-  alias ExPlasma.Transaction
+  @empty_metadata <<0::256>>
+  @empty_tx_data 0
+  @tx_type TypeMapper.tx_type_for(:tx_payment_v1)
+  @output_type TypeMapper.output_type_for(:output_payment_v1)
 
-  @type validation_responses() ::
-          ExPlasma.Output.Type.PaymentV1.validation_responses()
-          | {:error, {:inputs, :cannot_exceed_maximum_value}}
-          | {:error, {:outputs, :cannot_exceed_maximum_value}}
+  defstruct tx_type: @tx_type, inputs: [], outputs: [], tx_data: @empty_tx_data, metadata: @empty_metadata
 
-  # The maximum input and outputs the Transaction can have.
-  @output_limit 4
+  @type t() :: %__MODULE__{
+          tx_type: pos_integer(),
+          inputs: outputs(),
+          outputs: outputs(),
+          tx_data: any(),
+          metadata: metadata()
+        }
```

Comment:
perhaps this is a good oppurtunity to introduce `@enforce_keys [:mandatory]`? is this something we can use in ex_plasma as well?

Reply:
As discussed in the channel, I think we're good with using default for our case